logic/resolvable/Concludable.java

```diff
@@ -188,8 +188,13 @@ boolean unificationSatisfiable(ThingVariable concludableThingVar, ThingVariable
         }
 
         if (!concludableThingVar.value().isEmpty() && !conclusionThingVar.value().isEmpty()) {
-            // TODO: detect value contradictions between constant predicates
-            satisfiable &= true;
```

Comment:
(Entry point)
For a concludable to unify with a conclusion, the constraints on the conclusion are compatible with the constraints on the concludable.
This is currently only useful in the case where the conclusion is an explicit-has, where the concluded value can be known statically. In all other cases, the concluded value is variables and can thus satisfy any single constraint in isolation.

(We cannot consider two constraints at a time since concludables consist of only one constraint. Trying to support this is likely out of scope of this optimisation and what the reasoner does)